evalml/pipelines/components/transformers/preprocessing/datetime_featurizer.py

```diff
@@ -113,3 +113,11 @@ def get_feature_names(self):
            mapping the unique feature values to their integer encoding.
         """
         return self._categories
+
+    def _get_feature_provenance(self):
+        provenance = {}
+        for col_name in self._date_time_col_names:
+            provenance[col_name] = []
+            for feature in self.parameters['features_to_extract']:
+                provenance[col_name].append(f'{col_name}_{feature}')
```

Comment:
Good observation! I like the "from scratch" pattern because it keeps the definition of the provenance separate from the transform logic but you're right it can lead to some duplication. I think in most cases I tried to follow the "from scratch" pattern but I couldn't use it for all because the OHE, for example, has special logic for making names unique so the provenance basically has to happen in `get_feature_names`.